fuel-txpool/src/service.rs

```diff
@@ -168,19 +156,40 @@ pub mod tests {
         let tx1 = Arc::new(DummyDb::dummy_tx(tx1_hash));
         let tx2 = Arc::new(DummyDb::dummy_tx(tx2_hash));
 
-        let service = TxPoolService::new(db, config);
-        let out = service.insert(vec![tx1, tx2]).await;
+        let service = Service::new(db, config).unwrap();
+        service.start(br).await;
+
+        let (response, receiver) = oneshot::channel();
+        let _ = service
+            .sender()
+            .send(TxPoolMpsc::Insert {
+                txs: vec![tx1, tx2],
+                response,
+            })
+            .await;
+        let out = receiver.await.unwrap();
+
         assert_eq!(out.len(), 2, "Shoud be len 2:{:?}", out);
         assert!(out[0].is_ok(), "Tx1 should be OK, got err:{:?}", out);
         assert!(out[1].is_ok(), "Tx2 should be OK, got err:{:?}", out);
-        let out = service.find(&[tx1_hash, tx3_hash]).await;
+        let (response, receiver) = oneshot::channel();
+        let _ = service
+            .sender()
+            .send(TxPoolMpsc::Find {
+                ids: vec![tx1_hash, tx3_hash],
+                response,
+            })
+            .await;
+        let out = receiver.await.unwrap();
         assert_eq!(out.len(), 2, "Shoud be len 2:{:?}", out);
         assert!(out[0].is_some(), "Tx1 should be some:{:?}", out);
         let id = out[0].as_ref().unwrap().id();
         assert_eq!(id, tx1_hash, "Found tx id match{:?}", out);
         assert!(out[1].is_none(), "Tx3 should not be found:{:?}", out);
+        service.stop().await.unwrap().await.unwrap();
     }
 
+    /*
```

Comment:
Is this test blocked from working by something or no longer needed? I saw `Subscriber` was removed so if it's no longer needed I would suggest removing the test. If it is blocked by something could you make issue to track this for when it should be reinstated?

Reply:
I left it as maybe code that can be reused, the `subscription` is now done with `tokio::broadcast`. I will remove the comment.

Reply:
I feel like it's a smell to have commented out tests without any explanation in the code. If we want this code as a reference for later it should be put in a doc-test or something else to avoid bit-rot.

Reply:
test was rewriten